gateway/src/apicast/policy/keycloak_scope_check/apicast-policy.json

```diff
@@ -0,0 +1,63 @@
+{
+  "$schema": "http://apicast.io/policy-v1/schema#manifest#",
+  "name": "Keycloak Scope Check",
+  "summary": "Adds scope check with Keycloak.",
+  "description": [
+    "This policy adds scope check with Keycloak.\n",
+    "This policy exposes realm roles and client roles in the access token."
+  ],
+  "version": "builtin",
+  "configuration": {
+    "type": "object",
+    "properties": {
+      "scopes": {
+        "type": "array",
+        "items": {
+          "type": "object",
+          "properties": {
+            "realm_roles": {
+              "type": "array",
+              "description": "Realm roles",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "name": {
+                    "type": "string",
+                    "description": "Name of the role"
+                  }
+                }
+              }
+            },
+            "client_roles": {
+              "type": "array",
+              "description": "Client roles",
+              "items": {
+                "type": "object",
+                "properties": {
+                  "name": {
+                    "type": "string",
+                    "description": "Name of the role"
+                  },
+                  "client": {
+                    "type": "string",
+                    "description": "Client of the role"
+                  }
+                }
+              }
+            },
+            "resource": {
+              "type": "string",
+              "description": "Resource controlled by scope. Specify the path(like '/aaa') of the resource. Wildcard({wildcard}) can be used."
```

Comment:
I think this should say it is the same format as Mapping Rules. And that it matches from the beginning of the string and to make an exact match you need to use `$` at the end.

Reply:
Changed.